packages/cosmic-swingset/src/launch-chain.js

```diff
@@ -413,6 +414,19 @@ export async function launch({
 
   // Not to be confused with the gas model, this meter is for OpenTelemetry.
   const metricMeter = metricsProvider.getMeter('ag-chain-cosmos');
+
+  // Define a single counter for all action types with dimensional attributes
+  const actionCounter = metricMeter.createCounter('action_total', {
+    description: 'Total number of actions',
+  });
+
+  /** @type {Record<ActionType.QueuedActionType, (attributes?: Record<string, unknown>) => void>} */
+  const actionMetrics = Object.fromEntries(
+    Object.keys(ActionType.QueuedActionType).map(actionType => [
+      ActionType.QueuedActionType[actionType],
+      (attributes = {}) => actionCounter.add(1, { actionType, ...attributes }),
+    ]),
+  );
```

Comment:
Suggestions: use less generic names for these variables, and define the latter a bit more straightforwardly.
```suggestion
  const knownActionTypes = new Set(Object.values(ActionType.QueuedActionType));

  const processedInboundActionCounter = metricMeter.createCounter(
    'cosmic_swingset_inbound_actions',
    { description: 'Processed inbound action counts by type' },
  );

  /** @type {(actionType: ActionType.QueuedActionType) => void>} */
  const countInboundAction = actionType => {
    if (!knownActionTypes.has(actionType)) {
      console.warn(`unknown inbound action type ${JSON.stringify(queueName)}`);
    }
    processedInboundActionCounter.add(1, { actionType });
  };
```
I'd also like to see metric definitions consolidated in kernel-stats.js rather than scattered throughout the code, but maybe not yet to minimize conflicts with #10904. It probably should at least appear closer to the definition of `makeInboundQueueMetrics`, though.